src/Microsoft.Health.Dicom.Api.UnitTests/Features/Exceptions/ExceptionHandlingMiddlewareTests.cs

```diff
@@ -57,6 +57,8 @@ public static IEnumerable<object[]> GetExceptionToStatusCodeMapping()
         yield return new object[] { new TaskCanceledException(), HttpStatusCode.BadRequest };
         yield return new object[] { new InvalidOperationException(), HttpStatusCode.BadRequest };
         yield return new object[] { new PayloadTooLargeException(1), HttpStatusCode.RequestEntityTooLarge };
+        yield return new object[] { new DataStoreException(new Exception(), isExternal: true), HttpStatusCode.FailedDependency };
```

Comment:
Should we add tests that capture the `innerException` handling of BYOS exceptions as defined [here](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#bring-your-own-storage-byos)?

Reply:
Thanks for the link. I have in the BlobFileTests